src/main/scala/piconot/internal/InternalDSL.scala

```diff
@@ -0,0 +1,126 @@
+package piconot.internal
+import picolib.maze.Maze
+import picolib.semantics._
+import scala.collection.mutable.MutableList
+import java.io.File
+import scalafx.application.JFXApp
+
+/*
+ * Internal DSL implementation for creating Picobot rules.
+ * Despite great effort, we were unable to make an accurate implementation 
+ * of our language using Internal DSL methods. Though it may have been possible
+ * with intense pattern-matching, this would have been way too timely. We decided
+ * to make a version of our DSL that ended up being useful as an intermediate 
+ * representation for our External DSL. So not all was done in vain!
+ */
+class InternalDSL extends JFXApp {
+  
+  def setSurroundings (freeArray:Array[String], occArray:Array[String]) : Surroundings = { 
+    
+    // Define surroundings
+     var surr_array = new Array[RelativeDescription](4)
+     
+    // Initialize all indices to Anything
+     var x = 0
+     for (x<- 0 to (surr_array.size - 1)) {
+       surr_array.update(x, Anything)
+     }
+     
+    // Set N, E, W, or S to Open
+     if (freeArray.contains("N")) {
+       surr_array.update(0, Open)
+     }
+     if (freeArray.contains("E")) {
+       surr_array.update(1, Open)
+     }
+     if (freeArray.contains("W")) {
+       surr_array.update(2, Open)
+     }
+     if (freeArray.contains("S")) {
+       surr_array.update(3, Open)
+     }
+     
+     // Set N, E, W, or S to Blocked
+     if (occArray.contains("N")) {
+       surr_array.update(0, Blocked)
+     }
+     if (occArray.contains("E")) {
+       surr_array.update(1, Blocked)
+     }
+     if (occArray.contains("W")) {
+       surr_array.update(2, Blocked)
+     }
+     if (occArray.contains("S")) {
+       surr_array.update(3, Blocked)
+     }
```

Comment:
I like how nicely this takes care of potential conflicts where a direction is in both "free" and "occ" arrays - the `occArray` has precedence, which is probably the safe route.